katib-ui-rock/rockcraft.yaml

```diff
@@ -19,36 +19,86 @@ platforms:
   amd64:
 
 parts:
-  katib-ui:
+  frontend-lib:
+    plugin: nil
+    source: https://github.com/kubeflow/kubeflow
+    source-type: git
+    # Set source-tag here to the commit cited in https://github.com/kubeflow/katib/blob/master/pkg/ui/v1beta1/frontend/COMMIT
+    # for this Katib release
+    # Note: Katib 0.16.0 cannot build with Kubeflow v1.8-branch because Kubeflow refactored this lib
+    source-commit: 046c6d3c8301a95a2809c0d927bbbf72638dd639
```

Comment:
I think the end of your comment captures my worry about the beginning :D

It does feel odd that a user has to keep `frontend` and `frontend-lib`'s git checkouts in sync, especially since the `katib/katib` repo checked out by `frontend` includes that `COMMIT` file that specifies the commit hash we want to use from `kubeflow/kubeflow` when building `frontend-lib`.  We could remove that human-sync by taking the commit hash from the `COMMIT` file like you describe.

Where it gets awkward I think is that we'd have to pass it inside the rockcraft parts (since we can't dynamically write to `source-commit` in the rockcraft.yaml).  What we could do is, in `frontend-lib`, in a override-build do `commit=$(cat /root/parts/frontend/src/path/to/COMMIT); git checkout kubeflow/kubeflow $commit; ...`.  That would remove the human-sync effort, but make a more complicated rock file.  wdyt?

Reply:
Oh I didn't realise this is possible actually. Hmm, this feels a better approach (less error prone updates) but isn't weird/not recommended to access files from another part in terms `rockcraft` best practices? I 'm OK with the file as is for the reasons we both mentioned, so I 'll leave this up to you.

Reply:
Yeah it feels anti-rockcraft, but also handy.  But maybe too complex for the effort it saves? 

I'll leave it as is, but we could discuss with the team in future.  sound good?

Reply:
Sure, we could. We 'll figure this out for good in case someone else (apart from us) updates this for 1.9 release. Added `source-commit` in the [ROCK upgrade section](https://docs.google.com/document/d/1sCTnY3T4b6FEv5jmHlyB2PYC2ZGCY4tAB4cKTk-R7FY/edit#heading=h.u5ruz84494xq) in the spec